frodo.js

```diff
@@ -2,86 +2,86 @@
  * Extras Plugin
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.